core/authenticate/service.go

```diff
@@ -156,11 +166,43 @@ func (s Service) StartFlow(ctx context.Context, request RegistrationStartRequest
 		FinishURL: request.ReturnToURL,
 		CreatedAt: s.Now(),
 		ExpiresAt: s.Now().Add(defaultFlowExp),
+		Email:     request.Email,
 		Metadata: metadata.Metadata{
 			"callback_url": request.CallbackUrl,
 		},
 	}
 
+	if request.Method == PassKeyAuthMethod.String() {
+		newPassKeyUser := strategy.NewPassKeyUser(request.Email)
```

Comment:
`newPassKeyUser := strategy.NewPassKeyUser(request.Email)` we don't need to create it here, we can create this inside respective login/register methods

Reply:
passing request in place of newPassKeyUser in the function and creating user inside that function.